shared/crypto/keypair.go

```diff
@@ -36,18 +36,25 @@ type KeyPair interface {
 	// Marshalling
 	Marshal() ([]byte, error)
 	Unmarshal([]byte) error
+	MarshalJSON() ([]byte, error)
+	UnmarshalJSON([]byte) error
 }
 
 var _ KeyPair = &encKeyPair{}
 
 // encKeyPair struct stores the public key and the passphrase encrypted private key
 // The encrypted private key is stored as a JSON string with the fields needed to decrypt the key
 type encKeyPair struct {
-	PublicKey     PublicKey
-	PrivKeyArmour string
+	PublicKey     PublicKey `json:"public_key"`
+	PrivKeyArmour string    `json:"priv_key_armour"`
 }
 
-// newKeyPair Generates a new KeyPair interface given the public key and armoured private key
+type simpleEncKeyPair struct {
```

Comment:
I was looking around and found this: https://medium.com/picus-security-engineering/custom-json-marshaller-in-go-and-common-pitfalls-c43fa774db05

The tl;dr is that if we define a `func (pk *PublicKey) MarshalJSON() ([]byte, error) { ...}`, we'd be able to avoid the need of having `simpleEncKeyPair` altogether.

Cc @h5law  for visibility

Here's an example (https://go.dev/play/p/a5OojMD3atJ):

<img width="587" alt="Screenshot 2023-03-07 at 4 16 11 PM" src="https://user-images.githubusercontent.com/1892194/223585968-3977a7d8-c6ef-4643-abbf-481e2bf839a8.png">

Reply:
I racked my head against this a bit, I don't think there is a way to encode the public key as a string and unmarshal as a crypto.PublicKey without a temporary struct. Even that article uses that technique to unmarshal.

Though this new approach is much cleaner and simplified the interfaces a lot.

Reply:
@0xBigBoss Just to jump back in on this one - I thought that the previous comment I left on the function in `ed25519.go` that can generate a new public key from bytes would be the solution here. The marshalled `string` does not need to be human readable does it? If we just do `string(rawBz)`  then we can do `NewPublicKeyFromBytes([]byte(rawBzString))`?

I think this should work, the encoded string doesn't have any requirements on being human readable - nobody will be reading it anyway so there is not a need for it to "look nice".

Reply:
@h5law But don't we run into the same issue that gob uses it's own encoding so it's not easily consumed by other SDKs?

Being able to easily read the values from vault has come in handy before at least from my experience.

<img width="1724" alt="Screenshot 2023-03-08 at 16 41 56" src="https://user-images.githubusercontent.com/95193764/223867988-d29a9f85-f613-4b4c-ac69-00d38727ccf4.png">

Reply:
@0xBigBoss So `gob` encoding would be specific to golang, I'm pretty sure, so you are correct on that front but with JSON encoding to `[]byte` this would be consistent across all potential SDKs. I am not really too invested in either now as the idea of a Protobuf for the `KeyPair` would be the most portable option - serialising these would also not be human readable though. I think the human readable encoding should be left to exporting the key either in RAW or JSON format IMO.

The only thing is that introducing another struct to ease the encoding seems unnecessary to me, that's all really. If we can do the same without introducing the `simpleEncodedKey` then I think we should go with that option.

Reply:
The TLDR on why we need a temp/anon struct is cuz of the implementation `UnmarshalJSON` of the public key. We are assuming it's using hex encoding. Go json uses base64 encoding for byte[]. We'd have to change this implementation (which I don't recommend, base16 is common and expected). 

@Olshansk @h5law 

https://github.com/pokt-network/pocket/blob/main/shared/crypto/ed25519.go#L198-L214